phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -748,6 +757,14 @@ public Long getEstimateInfoTimestamp() throws SQLException {
         }
     }
 
+    /**
+     * Implementation of MutationPlan that is selected if
+     * 1) there are mutable indexes presented for the table,
+     * 2) auto commit is on,
+     * 3) the table is not transactional,
+     * 4) there is no WHERE clause, and
+     * 5) no LIMIT clause in the DELETE statement.
```

Comment:
every plan except `MultiRowDeleteMutationPlan` can have LIMIT and WHERE correct? 

i might have misread earlier, otherwise would have commented, sorry.

Reply:
basically, both `ServerSelectDeleteMutationPlan` and `ClientSelectDeleteMutationPlan` can have LIMIT and/or WHERE clause (for non-pk columns)

Reply:
If there is LIMIT clause, it will go client side

Reply:
i see, `boolean runOnServer = isAutoCommit && !hasPreOrPostProcessing && .....`
so `!hasPreOrPostProcessing` is applicable to both `MultiRowDeleteMutationPlan` and `ServerSelectDeleteMutationPlan`.

what this means is, if `hasPreOrPostProcessing` is true, only `ClientSelectDeleteMutationPlan` will get selected.

`hasPreOrPostProcessing` is always true for LIMIT clause, alright this one looks good, thanks